File: src/main/java/com/smartwallet/watchlist/entity/WatchlistItem.java
package com.smartwallet.watchlist.entity;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity(name = "LegacyWatchlistItem")
@Table(name = "watchlist_items")
public class WatchlistItem {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "watchlist_id", nullable = false)
    private Long watchlistId;
    
    @Column(name = "asset_symbol", nullable = false)
    private String assetSymbol;
    
    @Column(name = "position")
    private Integer position;
    
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Long getWatchlistId() { return watchlistId; }
    public void setWatchlistId(Long watchlistId) { this.watchlistId = watchlistId; }
    public String getAssetSymbol() { return assetSymbol; }
    public void setAssetSymbol(String assetSymbol) { this.assetSymbol = assetSymbol; }
    public Integer getPosition() { return position; }
    public void setPosition(Integer position) { this.position = position; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
}

File: worktree-pipeline/src/main/kotlin/com/smartwallet/watchlist/model/WatchlistItem.kt
package com.smartwallet.watchlist.model

import jakarta.persistence.*
import java.time.Instant

@Entity
@Table(name = "watchlist_items")
data class WatchlistItem(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "watchlist_id", nullable = false)
    val watchlist: Watchlist,

    @Column(name = "asset_symbol", nullable = false, length = 20)
    val assetSymbol: String,

    @Column
    val position: Int? = 0,

    @Column(name = "created_at")
    val createdAt: Instant? = null
)
